core/src/main/java/io/grpc/CallCredentials.java

```diff
@@ -65,6 +72,7 @@
    * @param applier The outlet of the produced headers. It can be called either before or after this
    *        method returns.
    */
+  @ExperimentalApi("https//github.com/grpc/grpc-java/issues/1914")
   void applyRequestMetadata(
```

Comment:
> If methods change, ... implementors of CallCredentials will not [change].

I don't want to provide that guarantee. And I don't see how that is true, unless you are saying there would be no implementors so it is vacuously true.

> As the code stands with this PR, someone extending CallCredentials, a stable class, will suffer breakage when experimental methods come and go. That is not stable.

It is impossible to extend CallCredentials without touching ExperimentalApi, so I feel like your sentence is a word-game that is skewing reality. Extending CallCredentials is not stable, because it is impossible with the current ExperimentalApi annotations and even explicitly mentioned in the documentation.

I'd be willing to make CallCredentials an abstract class or add a `doNotImplementThisOrYouWillBreak()`, but in either case I want to reserve the right to _completely_ change the contents, like adding a new abstract method and removing the existing method.

I could maybe be okay with leaving ExperimentalApi on the interface, but say that we guarantee its name as stable.